aiida/restapi/api.py

```diff
@@ -111,6 +113,14 @@ def __init__(self, app=None, **kwargs):
             resource_class_kwargs=kwargs
         )
 
+        self.add_resource(
+            QueryBuilder,
+            '/query/',
```

Comment:
I suggest `queryhelp` instead.

This is the term used in AiiDA for this JSON format, while "query" would apply to all current REST API endpoints

Reply:
I agree with this and will change it (also for the other comments), but I quite dislike the name `queryhelp` in general, because it doesn't reflect what it is - but that's maybe for another PR? Although it might be good to change the name _now_ as to not have to deprecate this immediately?

Suggested names for the `QueryBuilder.queryhelp`:
- `query_json` / `json_query` (or without the underscore, `_`)
- `query`
- `definition`
- `query_definition`

What do you think?

Reply:
I agree with @CasperWA on the name `queryhelp`. Most intuitive to me would be two methods:

* `QueryBuilder.as_dict()`: This has the same functionality as the current `queryhelp` property: it returns the `QueryBuilder` instance as a dictionary, which you can use to recreate the same instance with:
* `QueryBuilder.from_dict(d)`: This class method returns the `QueryBuilder` instance from the dictionary created by the `as_dict()` method. This basically just returns `QueryBuilder(**d)`.

I think this API is very clear, but that may be because I was searching for it due to my experience with the [`monty`](https://guide.materialsvirtuallab.org/monty/monty.json.html) package (heavily used in the `pymatgen`-branch of materials science packages).

Anyways, we probably should open an issue (or discussion? 😄 ) to discuss this.

Reply:
@mbercx if you want to add a PR that adds the `as_dict` function to the querybuilder, I'm happy to review it. I agree that this works well.

As for the REST API endpoint, what about `/qb` or `/querybuilder`?
The fact that we'll need a JSON representation of the querybuilder can be assumed explicitly, since we're anyhow outputting JSON.

P.S. By the way, it seems facebook is also lazy in updating their API docs...
![image](https://user-images.githubusercontent.com/1053245/96889987-78842a00-1487-11eb-84ae-a62b58c2282e.png)